// vue-marquee-slider/dist/vue-marquee-slider.ssr.js
'use strict';function _iterableToArrayLimit(arr, i) {
  var _i = null == arr ? null : "undefined" != typeof Symbol && arr[Symbol.iterator] || arr["@@iterator"];
  if (null != _i) {
    var _s,
      _e,
      _x,
      _r,
      _arr = [],
      _n = !0,
      _d = !1;
    try {
      if (_x = (_i = _i.call(arr)).next, 0 === i) {
        if (Object(_i) !== _i) return;
        _n = !1;
      } else for (; !(_n = (_s = _x.call(_i)).done) && (_arr.push(_s.value), _arr.length !== i); _n = !0);
    } catch (err) {
      _d = !0, _e = err;
    } finally {
      try {
        if (!_n && null != _i.return && (_r = _i.return(), Object(_r) !== _r)) return;
      } finally {
        if (_d) throw _e;
      }
    }
    return _arr;
  }
}
function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}
function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}
function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}
function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];
  return arr2;
}
function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}//
//
//
//
//
//
//

var script = {
  name: "MarqueeSlider",
  props: {
    autoWidth: {
      type: Boolean,
      default: function _default() {
        return false;
      }
    },
    id: {
      type: String,
      required: true,
      default: function _default() {
        return "id";
      }
    },
    paused: {
      type: Boolean,
      default: function _default() {
        return false;
      }
    },
    repeat: {
      type: Number,
      default: function _default() {
        return 10;
      }
    },
    reverse: {
      type: Boolean,
      default: function _default() {
        return false;
      }
    },
    space: {
      type: Number,
      default: function _default() {
        return 200;
      }
    },
    speed: {
      type: Number,
      default: function _default() {
        return 1500;
      }
    },
    width: {
      type: Number,
      default: function _default() {
        return 100;
      }
    }
  },
  data: function data() {
    return {
      container: null,
      containerWidth: 0,
      items: [],
      itemsLength: 0,
      itemsWidth: []
    };
  },
  computed: {
    style: function style() {
      return "\n        animation-duration: ".concat(this.speed, "ms;\n        animation-direction: ").concat(this.reverse ? "reverse" : "normal", ";\n        animation-play-state: ").concat(this.paused ? "paused" : "running", ";\n      ");
    }
  },
  mounted: function mounted() {
    this.setContainer();
    this.setItems();
    this.setItemsLength();
    this.calculateContainerWidth();
    this.setContainerWidth();
    this.cloneItems();
  },
  methods: {
    calculateContainerWidth: function calculateContainerWidth() {
      for (var index = 0; index < this.itemsLength; index++) {
        this.itemsWidth.push(this.items[index].offsetWidth);
        this.setItemSpace(index);
        if (this.autoWidth) {
          // if we are not using width specified by the user, we are setting the object-fit:contain for the images
          this.setImageObjectFit(index);
        } else {
          // if the user use the width property and want all items to be equal, the component will set the minWidth of the items
          this.setItemWidth(index);
        }
      }
      this.containerWidth = this.itemsWidth.reduce(function (a, b) {
        return a + b;
      }, 0);
    },
    cloneItems: function cloneItems() {
      var repeatCounter = this.getRepeatCounter();
      for (var index = 0; index < repeatCounter; index++) {
        this.container.appendChild(this.items[index].cloneNode(true));
      }
    },
    getRepeatCounter: function getRepeatCounter() {
      return this.items.length * this.repeat;
    },
    setItems: function setItems() {
      // get all childrens that will be put inside the slot of the component
      this.items = this.container.children;
    },
    setItemsLength: function setItemsLength() {
      this.itemsLength = this.items.length;
    },
    setItemSpace: function setItemSpace(index) {
      this.items[index].style.marginRight = "".concat(this.space, "px");
    },
    setItemWidth: function setItemWidth(index) {
      this.items[index].style.minWidth = "".concat(this.width, "px");
    },
    setImageObjectFit: function setImageObjectFit(index) {
      this.items[index].style.objectFit = "contain";
    },
    setContainer: function setContainer() {
      this.container = document.querySelector("#".concat(this.id, " .").concat(this.$style.marqueeSliderContainer));
    },
    setContainerWidth: function setContainerWidth() {
      if (this.autoWidth) {
        this.container.style.width = "".concat(this.itemsLength * (this.containerWidth / this.itemsLength + this.space), "px");
      } else {
        this.container.style.width = "".concat(this.itemsLength * (this.width + this.space), "px");
      }
    }
  }
};function normalizeComponent(template, style, script, scopeId, isFunctionalTemplate, moduleIdentifier /* server only */, shadowMode, createInjector, createInjectorSSR, createInjectorShadow) {
    if (typeof shadowMode !== 'boolean') {
        createInjectorSSR = createInjector;
        createInjector = shadowMode;
        shadowMode = false;
    }
    // Vue.extend constructor export interop.
    const options = typeof script === 'function' ? script.options : script;
    // render functions
    if (template && template.render) {
        options.render = template.render;
        options.staticRenderFns = template.staticRenderFns;
        options._compiled = true;
        // functional template
        if (isFunctionalTemplate) {
            options.functional = true;
        }
    }
    // scopedId
    if (scopeId) {
        options._scopeId = scopeId;
    }
    let hook;
    if (moduleIdentifier) {
        // server build
        hook = function (context) {
            // 2.3 injection
            context =
                context || // cached call
                    (this.$vnode && this.$vnode.ssrContext) || // stateful
                    (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext); // functional
            // 2.2 with runInNewContext: true
            if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
                context = __VUE_SSR_CONTEXT__;
            }
            // inject component styles
            if (style) {
                style.call(this, createInjectorSSR(context));
            }
            // register component module identifier for async chunk inference
            if (context && context._registeredComponents) {
                context._registeredComponents.add(moduleIdentifier);
            }
        };
        // used by ssr in case component is cached and beforeCreate
        // never gets called
        options._ssrRegister = hook;
    }
    else if (style) {
        hook = shadowMode
            ? function (context) {
                style.call(this, createInjectorShadow(context, this.$root.$options.shadowRoot));
            }
            : function (context) {
                style.call(this, createInjector(context));
            };
    }
    if (hook) {
        if (options.functional) {
            // register for functional component in vue file
            const originalRender = options.render;
            options.render = function renderWithStyleInjection(h, context) {
                hook.call(context);
                return originalRender(h, context);
            };
        }
        else {
            // inject component registration as beforeCreate hook
            const existing = options.beforeCreate;
            options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
        }
    }
    return script;
}function createInjectorSSR(context) {
    if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
        context = __VUE_SSR_CONTEXT__;
    }
    if (!context)
        return () => { };
    if (!('styles' in context)) {
        context._styles = context._styles || {};
        Object.defineProperty(context, 'styles', {
            enumerable: true,
            get: () => context._renderStyles(context._styles)
        });
        context._renderStyles = context._renderStyles || renderStyles;
    }
    return (id, style) => addStyle(id, style, context);
}
function addStyle(id, css, context) {
    const group = css.media || 'default' ;
    const style = context._styles[group] || (context._styles[group] = { ids: [], css: '' });
    if (!style.ids.includes(id)) {
        style.media = css.media;
        style.ids.push(id);
        let code = css.source;
        style.css += code + '\n';
    }
}
function renderStyles(styles) {
    let css = '';
    for (const key in styles) {
        const style = styles[key];
        css +=
            '<style data-vue-ssr-id="' +
                Array.from(style.ids).join(' ') +
                '"' +
                (style.media ? ' media="' + style.media + '"' : '') +
                '>' +
                style.css +
                '</style>';
    }
    return css;
}/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function __vue_render__() {
  var _vm = this;
  var _h = _vm.$createElement;
  var _c = _vm._self._c || _h;
  return _c('div', {
    class: _vm.$style.marqueeSlider,
    attrs: {
      "id": _vm.id
    }
  }, [_vm._ssrNode("<div" + _vm._ssrClass(null, _vm.$style.marqueeSliderContainer) + _vm._ssrStyle(null, _vm.style, null) + ">", "</div>", [_vm._t("default")], 2)]);
};
var __vue_staticRenderFns__ = [];

/* style */
var __vue_inject_styles__ = function __vue_inject_styles__(inject) {
  if (!inject) return;
  inject("data-v-3942214c_0", {
    source: ".src-components-marqueeSlider-3ljS{overflow:hidden}.src-components-marqueeSliderContainer-1UFI{width:100%;animation-name:src-components-animation-2PZg;animation-timing-function:linear;animation-iteration-count:infinite;display:flex}@keyframes src-components-animation-2PZg{0%{transform:translateX(0)}100%{transform:translateX(-100%)}}",
    map: undefined,
    media: undefined
  });
  Object.defineProperty(this, "$style", {
    value: {
      "marqueeSlider": "src-components-marqueeSlider-3ljS",
      "marqueeSliderContainer": "src-components-marqueeSliderContainer-1UFI",
      "animation": "src-components-animation-2PZg"
    }
  });
};
/* scoped */
var __vue_scope_id__ = undefined;
/* module identifier */
var __vue_module_identifier__ = "data-v-3942214c";
/* functional template */
var __vue_is_functional_template__ = false;
/* style inject shadow dom */

var __vue_component__ = /*#__PURE__*/normalizeComponent({
  render: __vue_render__,
  staticRenderFns: __vue_staticRenderFns__
}, __vue_inject_styles__, __vue_script__, __vue_scope_id__, __vue_is_functional_template__, __vue_module_identifier__, false, undefined, createInjectorSSR, undefined);
var __vue_component__$1 = __vue_component__;/* eslint-disable import/prefer-default-export */var components$1=/*#__PURE__*/Object.freeze({__proto__:null,VueMarqueeSlider:__vue_component__$1});// install function executed by Vue.use()
var install = function installVueMarqueeSlider(Vue) {
  Object.entries(components$1).forEach(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
      componentName = _ref2[0],
      component = _ref2[1];
    Vue.component(componentName, component);
  });
};var components=/*#__PURE__*/Object.freeze({__proto__:null,'default':install,VueMarqueeSlider:__vue_component__$1});// Attach named exports directly to plugin. IIFE/CJS will
// only expose one global var, with component exports exposed as properties of
// that global var (eg. plugin.component)
Object.entries(components).forEach(function (_ref) {
  var _ref2 = _slicedToArray(_ref, 2),
    componentName = _ref2[0],
    component = _ref2[1];
  if (componentName !== 'default') {
    install[componentName] = component;
  }
});module.exports=install;
